server/src/main/java/org/elasticsearch/action/admin/indices/validate/query/TransportValidateQueryAction.java

```diff
@@ -75,12 +80,18 @@ public TransportValidateQueryAction(
             transportService.getThreadPool().executor(ThreadPool.Names.SEARCH)
         );
         this.searchService = searchService;
+        this.remoteClusterService = transportService.getRemoteClusterService();
     }
 
     @Override
     protected void doExecute(Task task, ValidateQueryRequest request, ActionListener<ValidateQueryResponse> listener) {
         request.nowInMillis = System.currentTimeMillis();
         LongSupplier timeProvider = () -> request.nowInMillis;
+
+        // Use a supplier to create the ResolvedIndices instance as necessary since it is only used to build the index metadata map for
+        // the query rewrite context
+        Supplier<ResolvedIndices> resolvedIndicesSupplier = createResolvedIndicesSupplier(request);
```

Comment:
I'm still using a supplier here because the `ResolvedIndices` instance is only used (lazily) by the query rewrite context, so most of the time it's not necessary to create it. LMK if this supplier usage is a problem.

Reply:
Done: https://github.com/elastic/elasticsearch/pull/107075/commits/7fff51552e31340572b38aa835646f4ef7c815ba